python/ray/train/tests/test_torch_device_manager.py

```diff
@@ -0,0 +1,84 @@
+import pytest
+import torch
+
+import ray
+from ray.air._internal.device_manager import (
+    CUDATorchDeviceManager,
+    NPUTorchDeviceManager,
+    get_torch_device_manager,
+)
+from ray.air._internal.device_manager.npu import NPU_TORCH_PACKAGE_AVAILABLE
+from ray.cluster_utils import Cluster
+from ray.train import ScalingConfig
+from ray.train.torch import TorchTrainer
+
+if NPU_TORCH_PACKAGE_AVAILABLE:
+    import torch_npu  # noqa: F401
+
+
+@pytest.fixture
+def ray_1_node_1_gpu_1_npu():
+    cluster = Cluster()
+    cluster.add_node(num_cpus=4, num_gpus=1, resources={"NPU": 1})
+    ray.init(address=cluster.address)
+
+    yield
+
+    ray.shutdown()
+    cluster.shutdown()
+
+
+def test_cuda_devcie_manager(ray_2_node_2_gpu):
+    def train_fn():
+        assert isinstance(get_torch_device_manager(), CUDATorchDeviceManager)
+
+    trainer = TorchTrainer(
+        train_loop_per_worker=train_fn,
+        scaling_config=ScalingConfig(
+            num_workers=1, use_gpu=True, resources_per_worker={"GPU": 1}
+        ),
+    )
+
+    trainer.fit()
+
+
+def test_npu_device_manager(ray_2_node_2_npus):
```

Comment:
where is this 2 node 2 npus fixture defined?

Reply:
Previously defined in `confest.py`, it has now been moved to `test_torch_device_manager.py`.